docs/guidelines/git-github.md

```diff
@@ -0,0 +1,306 @@
+  - Add a comment (`-C`) to help identify the key's purpose. This will be added to the public key as generic text but will not have any functional impact. It will not be exposed on github.com. However, the purpose of a public key is to "share it", so use with caution (PII).
+
+Example; Creating a SSH key for the Github user `larskaare`
+
+```shell
+ssh-keygen -t ed25519 -f ~/.ssh/github_larskaare_1 -C "Github SSH auth key for machine 1"
+```
+
+Two files are created, one named `github_larskaare_1` and one named `github_larskaare_1.pub`. The file with the `.pub` extension contains the public part of the key.
+
+!!! Tip "Re-using keys?"
+
+    We advice on creating separate SSH keys for separate machines and devices and not to re-use the same key on them all.
+
+!!! Tip "Passphrases"
+
+    Store passphrases in a password manager.
+
+#### Configure SSH and adding the key to the key-agent
+
+Adding the generated SSH key to the `ssh-agent` gives you a secure repository for your private keys's passphrases. Adding keys and passphrases to the key agent eliminates the need to repeatedly enter the passphrase.
+
+Follow the [official documentation](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent) of and add the SSH key to the ssh-agent. Be aware of the operating system selector at the top of the page - it will give you instructions for **Mac**, **Windows** and **Linux**.
+
+#### Adding SSH config
+
+SSH uses a config file for it's configuration (this file is not used by Git). The user-specific file is usually stored in `~/.ssh/config` and should be updated prior to using the SSH key and the ssh-key agent. Consult the documentation of your SSH client for details. (Many use OpenSSH which have good manual pages - [OpenSSH Manual Pages](https://www.openssh.com/manual.html))
+
+A SSH config file with minimum set-up usually looks like this:
+
+```
+Host github.com
+    AddKeysToAgent yes
+    IdentitiesOnly=yes
+    IdentityFile ~/.ssh/github_larskaare_1
+```
+
+#### Adding the private SSH key to the SSH agent
+
+We assume that the `ssh-agent` is available and running.
+
+The following command will add the private part of the SSH key we generated to the ssh-agent
+
+```shell
+ssh-add ~/.ssh/github_larskaare_1
+```
+
+For MacOS we would typically add the passphrase to the keychain as well
+
+```shell
+ssh-add --apple-use-keychain ~/.ssh/github_larskaare_1
+```
+
+### Configure github.com to use our SSH key
+
+We now have a SSH key that we can use when communicating with github.com. To be able to use this key with Github we need to upload the public part of the key to github.com
+
+- Print the public key
+
+```shell
+cat ~/.ssh/github_larskaare_1.pub
+```
+
+- Copy the public key and add it as a new SSH Authentication key to your Github profile at [https://github.com/settings/keys](https://github.com/settings/keys)
```

Comment:
Had a doubt here about how much of the "key" I should copy, should I include the algorithm? only the key? the description? 🤔